src/arch/ia32/descriptor/gate/trap/configuration.rs

```diff
@@ -0,0 +1,150 @@
+//! A module containing the implementation and tests for the [`Configuration`]
+//! structure.
+use super::{GateSize, PrivilegeLevel};
+use bit_field::BitField;
+use core::fmt;
+
+/// A structure representing the bits 0 to 15 from a
+/// [TrapGate](super::TrapGate) structure.
+///
+/// It gathers the following fields:
+/// - Size (D)
+/// - Descriptor privilege level (DPL)
+/// - Present bit (P)
+#[derive(Copy, Clone)]
+pub struct Configuration(u16);
+
+/// The set of all field offsets for the [`Configuration`] structure.
+mod offset {
+    /// Offset of the size (D) bit within the
+    /// [Configuration](super::Configuration) structure.
+    pub const SIZE: usize = 11;
+
+    /// Bounds of the descriptor privilege level (DPL) bits within the
+    /// [Configuration](super::Configuration) structure.
+    pub mod privilege_level {
+        /// Lower bit offset.
+        pub const LOWER: usize = 13;
+        /// Upper bit offset.
+        pub const UPPER: usize = 14;
+    }
+
+    /// Offset of the present bit (P) within the
+    /// [Configuration](super::Configuration) structure.
+    pub const PRESENT: usize = 15;
+}
+
+impl Default for Configuration {
+    fn default() -> Self {
+        Configuration(0x0700)
```

Comment:
Here again, if you agree with specifying the bits set

Reply:
Done in ba5c2c8